processor.go

```diff
@@ -76,18 +88,26 @@ func NewProcessor[AC any, OC any, JC any](ac AC, states []State[AC, OC, JC], ser
 	}
 }
 
-// Return is a struct that contains a job and a list of kick requests
-// that is used for returning job updates to the system
-type Return[JC any] struct {
-	Job          Job[JC]
-	KickRequests []KickRequest[JC]
-	Error        error
+func (p *Processor[AC, OC, JC]) validate(states []State[AC, OC, JC]) error {
```

Comment:
I was considering making States an object / typedef on an array of states (there's some weirdness with generics) that a function like this would live on. That way you could also work with state arrays (essentially the workflow) outside of a processor.. Also this lets you validate they're wrong earlier in your code flow.'

Really if feels like we access the states in a workflow in a few different ways, hash and array like at minimum.

Reply:
fixed in v2